src/conversion/viewconversiondispatcher.js

```diff
@@ -188,11 +203,23 @@ export default class ViewConversionDispatcher {
 	 * @see module:engine/conversion/viewconversiondispatcher~ViewConversionApi#convertChildren
 	 */
 	_convertChildren( input, consumable, additionalData = {} ) {
+		// Get all children of view input item.
 		const viewChildren = Array.from( input.getChildren() );
-		const convertedChildren = viewChildren.map( ( viewChild ) => this._convertItem( viewChild, consumable, additionalData ) );
 
-		// Flatten and remove nulls.
-		return convertedChildren.reduce( ( a, b ) => b ? a.concat( b ) : a, [] );
+		// 1. Map those children to model.
+		// 2. Filter out wrong results.
+		// 3. Extract children from document fragments to flatten results.
+		const convertedChildren = viewChildren
+			.map( ( viewChild ) => this._convertItem( viewChild, consumable, additionalData ) )
+			.filter( ( converted ) => converted instanceof ModelNode || converted instanceof ModelDocumentFragment )
```

Comment:
This is like a black hole. Can't we just throw here? Or at least warn the developer. But I think that throwing is better because this isn't even a runtime error but simply a serious implementation bug. The same applies to `_convertItem()` – I feel that we could throw there as well.